pkgs/development/python-modules/django-currentuser/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, django
+, hatchling
+}:
+let
+  version = "0.6.1";
+in
+buildPythonPackage {
+  pname = "django-currentuser";
+  inherit version;
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "zsoldosp";
+    repo = "django-currentuser";
+    rev = "v${version}";
+    hash = "sha256-sxt4ZMkaFANINd1faIA5pqP8UoDMXElM3unsxcJU/ag=";
+  };
+
+  disabled = pythonOlder "3.8";
+
+  nativeBuildInputs = [ hatchling ];
+
+  propagatedBuildInputs = [ django ];
+
+  # Running `python manage.py test testapp` as in `make test` fails:
+  # ModuleNotFoundError: No module named 'testapp'
+  doCheck = false;
```

Comment:
you can use execute django tests tests/setup.py instead

Reply:
Thanks for the suggestion! I've now tried many combinations of `python -m django`, `manage.py`, in the root directory, in the `tests` directory, running `setup.py`, etc. Nothing seems to work.
I keep getting `ModuleNotFoundError: No module named 'testapp'` originating from [`settings.py`](https://github.com/zsoldosp/django-currentuser/blob/ee77dc6ebbc34d1e853de19248d02c991f1c60a5/settings.py#L54). Could I ask you to give more precise commands to try (or otherwise approve the PR)?

Reply:
```
checkPhase = ''
    runHook preCheck
    ${python.interpreter} -m django check --settings=test_settings # or django check setup.py or something similar
    runHook postCheck
  '';
```

Reply:
There is no specific `settings.py` for tests, just the base one. I've never seen `setup.py` used for tests so I didn't think to try it, but in any case I can't get passed `ModuleNotFoundError: No module named 'testapp'`.

Reply:
@imincik Just wondering if you might have seen this kind of problem before?

Reply:
Upstream is using tox to run the tests in [their CI](https://github.com/zsoldosp/django-currentuser/blob/master/.github/workflows/tox.yml)

Reply:
PYTHONPATH=test tox --skip-missing-interpreters

Reply:
If you do `doInstallCheck` instead of `doCheck` you wouldn't need to set `PYTHONPATH`.

Reply:
Please set `env.DJANGO_SETTINGS_MODULE = settings;`.
The tox.ini will tell you how to run the test.

Reply:
Thanks! I looked at `tox.ini` and it runs `pip install -e tests`; I don't see an example in Nixpkgs of people using `pip` to run tests. Is this normal? How should I proceed?
Are there any docs about this, or about the `env.DJANGO_SETTINGS_MODULE` thing?

Reply:
That command just installs dependencies.
You will notice that you should read the makefile from the line after that.

Then please read the django documentation for more information about django.
I've never used it, so I don't know anything about it.
https://docs.djangoproject.com/en/5.0/topics/settings/

Reply:
Solved by adding `PYTHONPATH=tests:$PYTHONPATH` to `preCheck`. Thanks, @imincik!

Reply:
thats a helpful tip, thanks!